### PR TITLE
Include nitrate aerosol photolysis in the troposphere

### PR DESCRIPTION
**Pull request by viral211**
Include nitrate aerosol photolysis in the troposphere as described in:

> Shah, V., Jacob, D. J., Dang, R., Lamsal, L. N., Strode, S. A., Steenrod, S. D., Boersma, K. F., Eastham, S. D., Fritz, T. M., Thompson, C., Peischl, J., Bourgeois, I., Pollack, I. B., Nault, B. A., Cohen, R. C., Campuzano-Jost, P., Jimenez, J. L., Andersen, S. T., Carpenter, L. J., Sherwen, T., and Evans, M. J.: Nitrogen oxides in the free troposphere: implications for tropospheric oxidants and the interpretation of satellite NO2 measurements, Atmospheric Chemistry and Physics, 23, 1227–1257, https://doi.org/10.5194/acp-23-1227-2023, 2023.

### CODE REVIEW

**Comment by viral211** [comment]
Closes #1622

File: GeosCore/fast_jx_mod.F90 [hunk 1]
```diff
@@ -5324,23 +5333,41 @@ SUBROUTINE PHOTRATE_ADJ( Input_Opt, State_Diag, State_Met,                 &
...
     ! For all mechanisms. Set the photolysis rate of NITs and NIT to a
     ! scaled value of JHNO3. NOTE: this is set in geoschem_config.yml
-    IF ( Input_Opt%hvAerNIT ) THEN
+    ! Allow particulate nitrate photolysis in the troposphere only
+    IF ( Input_Opt%hvAerNIT .and.
+         State_Met%InTroposphere(I,J,L) ) THEN
+
+       ! Get NIT and NITs concentrations [molec cm-3]
+       C_NIT      = State_Chm%Species(id_NIT)%Conc(I,J,L)
+       C_NITs     = State_Chm%Species(id_NITs)%Conc(I,J,L)
```
Comment (yantosca) [inline GeosCore/fast_jx_mod.F90:5342]: It looks like C_NITs isn't used here

File: GeosCore/fast_jx_mod.F90 [hunk 2]
```diff
@@ -5311,6 +5319,7 @@ SUBROUTINE PHOTRATE_ADJ( Input_Opt, State_Diag, State_Met,                 &
 !
     REAL(fp) :: C_O2,     C_N2, C_H2,   ITEMPK, RO1DplH2O
     REAL(fp) :: RO1DplH2, RO1D, NUMDEN, TEMP,   C_H2O
+    REAL(fp) :: C_NIT,     C_NITs,    C_SALA,  C_SALC,  FAC
```
Comment (yantosca) [inline GeosCore/fast_jx_mod.F90:5322]: Would recommend making these pointers rather than scalars, as that would repeated copy operations.

**Review by yantosca — APPROVED** [review]
I've noted a couple of minor points on the PR but I can fix these and merge this up to 14.1.0 in preparation for inclusion into 14.2.0.

File: GeosCore/fast_jx_mod.F90 (hunk 2 above)
Comment (yantosca) [reply GeosCore/fast_jx_mod.F90:5322]: Actually I've left the C_NIT and C_SALA as scalars. There were a lot of unused variables so I removed them.

**Comment by yantosca** [comment]
Integration tests are in progress.

**Comment by yantosca** [comment]
Integration tests are running

**Comment by yantosca** [comment]
All GCHP integration tests passed
```console
==============================================================================
GCHP: Execution Test Results

Number of execution tests: 3
==============================================================================
 
Execution tests:
------------------------------------------------------------------------------
gchp_fullchem_benchmark_merra2_c48..................Execute Simulation....PASS
gchp_fullchem_standard_merra2_c24...................Execute Simulation....PASS
gchp_TransportTracers_geosfp_c24....................Execute Simulation....PASS
 
Summary of execution test results:
------------------------------------------------------------------------------
Execution tests passed:        3
Execution tests failed:        0
Execution tests not completed: 0

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%  All execution tests passed!  %%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
```

**Pull request merged**